src/backend/utils/adt/dbsize.c

```diff
@@ -180,9 +180,7 @@ calculate_database_size(Oid dbOid)
 	Assert(Gp_role != GP_ROLE_EXECUTE);
 
 	if (dbOid == HcatalogDbOid)
-		ereport(ERROR,
-			(ERRCODE_UNDEFINED_DATABASE,
-			errmsg("database hcatalog (OID 6120) is reserved")));
+		return -1;
```

Comment:
Should we return -1 ? For other non existent tables we return ERROR:  database <dbname> does not exist.
Shouldn't we return an actual error instead of -1 ?